### ISSUE TITLE
Indexing outside bounds buggy

### ISSUE THREAD

**Issue by FilipeMaia** [FilipeMaia/afnumpy#15]
When using slices larger than the array it raises an exception when it shouldn't.
When using arrays with elements larger than the array it returns non sensical values.

Examples:

``` python
import afnumpy

a = afnumpy.zeros((5))
a[1:8]
/Library/Python/2.7/site-packages/arrayfire-3.3.0-py2.7.egg/arrayfire/array.pyc in __getitem__(self, key)
    992             return out
    993         except RuntimeError as e:
--> 994             raise IndexError(str(e))
    995
    996

IndexError: ('In function dim_t af::calcDim(const af_seq &, const dim_t &)\nIn file src/backend/dim4.cpp:206\nInvalid dimension for argument 1\nExpected: seq.end < parentDim\n', 203)
```

``` python
import afnumpy

a = afnumpy.zeros((5))
print a[afnumpy.array([9])]
array([ 0.])
```

**Comment by FilipeMaia** [comment]
The first issue is fixed by b7b677070e191b608c396aebd7393af1256f919c.

**Comment by FilipeMaia** [comment]
The second one is solved by 52c2216cb626efc6068caebb3274afa9ee99d59c.
